### PR TITLE
Support Bigquery Views

### PR DESCRIPTION
**Pull request by jimmy-btn**
Extend the Google provider for Terraform to support Bigquery Views.

Make a few drive by cleanups in the process:

Amend the Basic Table test to actually verify that a valid table has been created
Support modifying Table Schema by parsing computed IDs correctly
Assign a default location to Tables created before Bigquery supported two zones

### CODE REVIEW

File: google/resource_bigquery_table.go [hunk 1]
```diff
@@ -394,3 +421,15 @@ func flattenTimePartitioning(tp *bigquery.TimePartitioning) []map[string]interfa
 
 	return []map[string]interface{}{result}
 }
+
+func expandView(configured interface{}) *bigquery.ViewDefinition {
+	raw := configured.([]interface{})[0].(map[string]interface{})
```
Comment (mightyguava) [inline google/resource_bigquery_table.go]: This is weird, is this going to panic if someone specifies an empty list in the config?

File: google/resource_bigquery_table.go [hunk 2]
```diff
@@ -92,6 +92,23 @@ func resourceBigQueryTable() *schema.Resource {
 				},
 			},
 
+			// View: [Experimental] If specified, configures this table as a view.
+			"view": &schema.Schema{
+				Type:     schema.TypeList,
```
Comment (mightyguava) [inline google/resource_bigquery_table.go]: So to specify nested objects in terraform configs, you make a list of 1?

Comment (jimmy-btn) [inline google/resource_bigquery_table.go]: Yup - not the way I would have done it, but for some reason the Schema DSL used by Terraform does not support fields of type interface{}, only lists of interface{}.

The time_partitioning field uses this same approach, so it's at least idiomatic, if not entirely sane.

File: google/resource_bigquery_table.go (hunk 1 above)
Comment (jimmy-btn) [inline google/resource_bigquery_table.go]: It absolutely would - but that's not possible in the DSL (HCL) today.

Here's HCL for a single view:

view {
  query: "Whatever"
}

and here is the HCL for a list of views:

view {
  query: "Whatever"
}

view {
  query: "Whatever"
}

**Pull request merged**